plugin-server/src/main/ingestion-queues/session-recording-v2/sessions/s3-session-batch-writer.ts

```diff
@@ -9,11 +9,18 @@ import { SessionBatchFileStorage, SessionBatchFileWriter, WriteSessionResult } f
 class S3SessionBatchFileWriter implements SessionBatchFileWriter {
     private stream: PassThrough
     private uploadPromise: Promise<CompleteMultipartUploadCommandOutput>
-    private writeError: Promise<never>
     private key: string
     private currentOffset = 0
+    private timeoutId: NodeJS.Timeout | null = null
+    private error: Error | null = null
+    private rejectCallbacks: ((error: Error) => void)[] = []
 
-    constructor(private readonly s3: S3Client, private readonly bucket: string, private readonly prefix: string) {
+    constructor(
+        private readonly s3: S3Client,
+        private readonly bucket: string,
+        private readonly prefix: string,
+        private readonly timeout: number = 5000
```

Comment:
Rather leave the default to the config (otherwise it makes it look like there are two places you can set the default which is misleading